src/libs/actions/Report.ts

```diff
@@ -431,6 +421,47 @@ function addActions(reportID: string, text = '', file?: File) {
         DateUtils.setTimezoneUpdated();
     }
 
+    const loginList = ReportUtils.getMentionListFromComment((reportCommentAction?.message ?? [])[0].html ?? '');
+
+    // When we send a mention in the chat room, we will invite this user to the chat room if this user isn't a participant of the room.
+    // So we should update participant of the report in optimistic data.
+    if (loginList.length > 0 && ReportUtils.isUserCreatedPolicyRoom(report as OnyxEntry<Report>)) {
+        const participantAccountIDs = report?.participantAccountIDs ?? [];
+        const accountIDs = PersonalDetailsUtils.getAccountIDsByLogins(loginList);
+        const newAccountIDs = accountIDs.filter((accountID) => !participantAccountIDs.includes(accountID));
+
+        if (newAccountIDs.length > 0) {
+            optimisticReport.participantAccountIDs = [...participantAccountIDs, ...newAccountIDs];
+            failureReport.participantAccountIDs = participantAccountIDs;
+
+            type PersonalDetailsOnyxData = {
+                optimisticData: OnyxUpdate[];
+                successData: OnyxUpdate[];
+                failureData: OnyxUpdate[];
+            };
+
+            const newPersonalDetailsOnyxData = PersonalDetailsUtils.getNewPersonalDetailsOnyxData(loginList, accountIDs) as PersonalDetailsOnyxData;
```

Comment:
why do we need the `as PersonalDetailsOnyxData`

I think we should set a return type for `getNewPersonalDetailsOnyxData()` instead